tests/producer_split_at.rs

```diff
@@ -342,3 +342,58 @@ fn zip() {
     check(&v[..5], || (0..5).into_par_iter().zip(10..20));
     check(&v[..5], || (0..10).into_par_iter().zip(10..15));
 }
+
+#[test]
+fn collect_drop_on_unwind() {
```

Comment:
This `producer_split_at.rs` has a collection of similar structured tests of producers, which doesn't seem to fit your test. Can you move it to its own file?